apps/api/src/domain/models/photograph/commands/create-photograph/photograph-created.event-handler.ts

```diff
@@ -0,0 +1,23 @@
+import { Inject } from '@nestjs/common';
+import { PhotographViewModel } from '../../../../../queries/buildViewModelForResource/viewModels/photograph.view-model';
+import { CoscradEventConsumer, ICoscradEventHandler } from '../../../../common';
+import { IPhotographQueryRepository, PHOTOGRAPH_QUERY_REPOSITORY_TOKEN } from '../../queries';
+import { PhotographCreated } from './photograph-created.event';
+
+@CoscradEventConsumer('PHOTOGRAPH_CREATED')
+export class PhotographCreatedEventHandler implements ICoscradEventHandler {
+    constructor(
+        @Inject(PHOTOGRAPH_QUERY_REPOSITORY_TOKEN)
+        private readonly photographRepository: IPhotographQueryRepository
+    ) {}
+
+    async handle(event: PhotographCreated): Promise<void> {
+        const { meta: { contributorIds = [] } = { contributorIds: [] } } = event;
+
+        const photograph = PhotographViewModel.fromPhotographCreated(event);
+
+        await this.photographRepository.create(photograph);
+
+        await this.photographRepository.attribute(photograph.id, contributorIds);
```

Comment:
I don't want to get too far before we we look at doing this atomically. I realize the views are eventually consistent, but I want to avoid the case where we create a view for a resource but fail to write the contribution.